mongo/change_stream.go

```diff
@@ -699,6 +739,18 @@ func (cs *ChangeStream) next(ctx context.Context, nonBlocking bool) bool {
 }
 
 func (cs *ChangeStream) loopNext(ctx context.Context, nonBlocking bool) {
+	if !validChangeStreamTimeouts(ctx, cs) {
+		cs.err = fmt.Errorf("maxAwaitTimeMS cannot be greater than or equal to timeoutMS")
```

Comment:
Optional: This error message is somewhat misleading because `validChangeStreamTimeouts` checks either the client-level timeout (i.e. `timeoutMS`) or the Context timeout. Also, the only way to set `maxAwaitTimeMS` is using the `MaxAwaitTime` option, so that is a better identifier to use. Consider a clearer error message:
```suggestion
		cs.err = fmt.Errorf("MaxAwaitTime must be less than the operation timeout")
```